godtools/App/Share/Data/UserDetailsRepository/Cache/RealmUserDetailsCache.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  RealmUserDetailsCache.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 11/21/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+import Combine
+
+class RealmUserDetailsCache {
+    
+    private let realmDatabase: RealmDatabase
+    private let userDetailsSync: RealmUserDetailsCacheSync
+    
+    init(realmDatabase: RealmDatabase, userDetailsSync: RealmUserDetailsCacheSync) {
+        
+        self.realmDatabase = realmDatabase
+        self.userDetailsSync = userDetailsSync
+    }
+    
+    func getUserDetailsChanged() -> AnyPublisher<Void, Never> {
+        
+        return realmDatabase.openRealm().objects(RealmUserDetails.self)
+            .objectWillChange
+            .eraseToAnyPublisher()
+    }
+    
+    func getUserDetails() -> UserDetailsDataModel? {
+        guard let realmUserDetails = realmDatabase.openRealm().objects(RealmUserDetails.self).first else { return nil }
```

Comment:
I did previously have the user ID in the keychain as part of the primary key that was used to retrieve the token.  But the issue with that was that later I didn't necessarily know what the user ID was in order to use it to retrieve the auth token.  So, I took it out of the primary key under the assumption that we should really only ever have 1 auth token in the keychain.  And I would think same goes from UserDetail objects-- we should realistically only ever have one in Realm, right?

You're right though that it would still be safer to retrieve the UserDetails using the ID (even if there's only 1 object in Realm).  How do you feel about having the user ID stored in the keychain separately from the auth token?

Reply:
Hmm I do like the idea of using the user ID as a primary key for fetching the auth token too.  It feels safer to have a primary key for fetching those.  And I think storing that user ID separately is a good idea.  If a user were to sign out we could always do some clean up by deleting their auth token / user ID from the Keychain.  I think the UserDetails we can always leave in the RealmDatabase.  If a user were to sign in with multiple accounts on their device we would then have that data persisted and available.

Reply:
a potential flow that could cause more than 1 `UserDetail` object to be present in realm:
1. User logs into app as User A
2. User realizes they used the wrong account and logs out
3. User then logs into app as User B

there would be multiple `UserDetail` objects in realm then unless you delete user related data on logout (which could be flaky)

Reply:
Sounds good!  Yeah I was assuming we'd probably want to clear everything out on sign out.  But if Okta logs out automatically sometimes that could be harder to track.

Reply:
yeah, and with us moving to support other authentication mechanisms, I'm not sure what they would support for callbacks